src/Native/Unix/Common/pal_safecrt.h

```diff
@@ -36,32 +41,61 @@ inline static bool add_s(T a, T b, T* result)
 #endif
 }
 
+#else // __cplusplus
+
+// Multiplies a and b into result.
+// Returns true if safe, false if overflows.
+inline static int8_t multiply_s(size_t a, size_t b, size_t* result)
```

Comment:
Should these rather return `bool` ?

Reply:
`bool` doesn't exist with C99, so the usual replacement is `int8_t`. Maybe we could add a `typedef int8_t bool`.

Reply:
`#include <stdbool.h>` ?